### PR TITLE
Extending doi handler for any html url

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

**Review by brinxmat — APPROVED** [review]

File: create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/MainHandler.java [hunk 1]
```diff
@@ -57,13 +62,18 @@
     private final transient BareProxyClient bareProxyClient;
     private final transient String publicationApiHost;
     private final transient String publicationApiScheme;
+    private final transient MetadataService metadataService;
 
     private static final Logger logger = LoggerFactory.getLogger(MainHandler.class);
 
+    /**
```
Comment (axthosarouris) [inline create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/MainHandler.java:69]: Update checkstyle config than adding non-informative javadoc

File: create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/MainHandler.java [hunk 2]
```diff
@@ -57,13 +62,18 @@
...
+     * Default constructor for MainHandler.
+     */
     @JacocoGenerated
     public MainHandler() {
         this(JsonUtils.objectMapper, new PublicationConverter(), new DoiTransformService(),
```
Comment (axthosarouris) [inline create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/MainHandler.java:74]: After the PR is merged, can you remove the objectMapper parameter? We use only one value here.

File: create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/MainHandlerTest.java [hunk 3]
```diff
@@ -115,13 +116,46 @@ public void testOkResponse()
...
+            throws Exception {
+        MainHandler mainHandler = createMainHandler(environment);
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        mainHandler.handleRequest(nonDoiUrlInputStream(), output, getMockContext());
+        GatewayResponse<Summary> gatewayResponse = parseSuccessResponse(output.toString());
+        assertEquals(SC_OK, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(MainHandler.ACCESS_CONTROL_ALLOW_ORIGIN));
+        Summary summary = gatewayResponse.getBodyObject(Summary.class);
+        assertNotNull(summary.getIdentifier());
```
[the remark this answers is not in the document]
Comment (axthosarouris) [reply create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/MainHandlerTest.java:130]: the context can be a field.

File: create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/MainHandlerTest.java [hunk 4]
```diff
@@ -115,13 +116,46 @@ public void testOkResponse()
...
+    }
+
+    @Test
+    public void testBadGatewayResponseWhenUrlIsInvalidNonDoi() throws Exception {
```
Comment (axthosarouris) [inline create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/MainHandlerTest.java:134]: refactor the setup as above

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 5]
```diff
@@ -29,20 +27,27 @@
...
     public void getMetadataJsonReturnsCreatePublicationRequest()
-            throws ExtractionException, IOException, URISyntaxException {
-        String json = createFramedPublicationJsonld();
-        CreatePublicationRequest request = MetadataConverter.fromJsonLd(json);
+            throws IOException {
+        MetadataService metadataService = new MetadataService();
+        URI uri = prepareWebServerAndReturnUri();
+        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
+
+        assertRequest(request);
```
Comment (axthosarouris) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:41]: assertThatRequestHasPublicationRequestWithAllNecessaryFieldsForSomeUnknownOperation

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 6]
```diff
@@ -51,22 +56,14 @@ public void getMetadataJsonReturnsCreatePublicationRequest()
 
         PublicationDate date = entityDescription.getDate();
         assertThat(date.getYear(), is(notNullValue()));
-
-        wireMockServer.stop();
     }
 
-    private String createFramedPublicationJsonld() throws IOException, ExtractionException, URISyntaxException {
+    private URI prepareWebServerAndReturnUri() {
```
Comment (axthosarouris) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:61]: uri of what?

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**